specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+      header={header}
+      headerButtons={
+        <>
+          <span className="-ml-2 flex-1" />
+          <Button.Small
+            aria-pressed={isEditing}
+            className="w-fit"
+            disabled={!hasPermission('/schemaconfig/uniquenessrules', 'update')}
+            onClick={toggleIsEditing}
+          >
+            {commonText.edit()}
+          </Button.Small>
+        </>
+      }
+      modal={false}
+      onClose={handleClose}
+    >
+      <table className="grid-table grid-cols-[2fr_1fr] gap-2 gap-y-4 overflow-auto ">
+        <thead>
+          <tr>
+            <td>{schemaText.uniqueFields()}</td>
+            <td>{schemaText.scope()}</td>
+          </tr>
+        </thead>
+        {tableRules?.map(({ rule, duplicates }, index) => (
+          <UniquenessRuleRow
+            fetchedDuplicates={duplicates}
+            fields={fields}
+            isEditing={isEditing}
+            key={index}
+            label={getUniqueInvalidReason(
+              rule.scopes.map(
+                (scope) =>
+                  getFieldsFromPath(model, scope).at(-1) as Relationship
+              ),
+              filterArray(rule.fields.map((field) => model.getField(field)))
+            )}
+            model={model}
+            relationships={relationships}
+            rule={rule}
+            onChange={(newRule): void => handleRuleValidation(newRule, index)}
+            onRemoved={(): void => setTableRules(removeItem(tableRules, index))}
+          />
+        ))}
+      </table>
+      {isEditing && (
+        <Button.Small
+          className="w-fit"
+          disabled={!hasPermission('/schemaconfig/uniquenessrules', 'create')}
+          onClick={(): void =>
+            handleRuleValidation(
+              {
+                id: null,
+                modelName: model.name,
+                fields: [fields[0].name],
+                isDatabaseConstraint: false,
+                scopes: [],
+              },
+              tableRules.length
+            )
+          }
+        >
+          {schemaText.addUniquenessRule()}
+        </Button.Small>
+      )}
+    </Dialog>
+  );
+}
+
+function UniquenessRuleRow({
+  rule,
+  model,
+  label,
+  fields,
+  relationships,
+  fetchedDuplicates,
+  isEditing,
+  onChange: handleChanged,
+  onRemoved: handleRemoved,
+}: {
+  readonly rule: UniquenessRule;
+  readonly model: SpecifyModel;
+  readonly label: string;
+  readonly fields: RA<LiteralField>;
+  readonly relationships: RA<Relationship>;
+  readonly fetchedDuplicates: UniquenessRuleValidation | undefined;
+  readonly isEditing: boolean;
+  readonly onChange: (newRule: typeof rule) => void;
+  readonly onRemoved: () => void;
+}): JSX.Element {
+  const readOnly = React.useMemo(
+    () =>
+      rule.isDatabaseConstraint ||
+      !hasPermission('/schemaconfig/uniquenessrules', 'update'),
+    [rule.isDatabaseConstraint]
+  );
+
+  const [
+    isModifyingRule,
+    _setModifyingRule,
+    _setNotModifyingRule,
```

Comment:
(optional) this is how we commonly handle this:

```suggestion
    _,
    __,
```

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/43aee5b97e52edca55b58e0cb2b76773754b510f